internal/controller/ocivalidator_controller.go

```diff
@@ -68,35 +68,43 @@ func (r *OciValidatorReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		Namespace: req.Namespace,
 	}
 	if err := r.Get(ctx, nn, vr); err == nil {
-		res, err := vres.HandleExistingValidationResult(nn, vr, r.Log)
-		if res != nil {
-			return *res, err
-		}
+		vres.HandleExistingValidationResult(nn, vr, r.Log)
 	} else {
 		if !apierrs.IsNotFound(err) {
 			r.Log.V(0).Error(err, "unexpected error getting ValidationResult", "name", nn.Name, "namespace", nn.Namespace)
 		}
-		res, err := vres.HandleNewValidationResult(r.Client, constants.PluginCode, nn, vr, r.Log)
-		if res != nil {
-			return *res, err
+
+		if err := vres.HandleNewValidationResult(
+			r.Client, constants.PluginCode, validator.Spec.ResultCount(), nn, r.Log,
+		); err != nil {
+			return ctrl.Result{}, err
 		}
 	}
 
-	failed := &types.MonotonicBool{}
-
-	// Registry Auth rules
-	for _, rule := range validator.Spec.RegistryAuthRules {
-		registryAuthService := registryauth.NewRegistryAuthRuleService(
+	// OCI Registry rules
+	for _, rule := range validator.Spec.OciRegistryRules {
+		ociRuleService := oci.NewOciRuleService(
 			r.Log,
 		)
-		validationResult, err := registryAuthService.ReconcileRegistryAuthRule(rule)
+		validationResult, err := ociRuleService.ReconcileOciRegistryRule(rule)
 		if err != nil {
-			r.Log.V(0).Error(err, "failed to reconcile Registry Auth rule")
+			r.Log.V(0).Error(err, "failed to reconcile OCI Registry rule")
 		}
-		vres.SafeUpdateValidationResult(r.Client, nn, validationResult, failed, err, r.Log)
+		vres.SafeUpdateValidationResult(r.Client, nn, validationResult, err, r.Log)
 	}
 
-	// TODO: execute more validation rules here
+	// ECR Registry rules
+	for _, rule := range validator.Spec.EcrRegistryRules {
+
+		ecrRuleService := ecr.NewEcrRuleService(
+			r.Log,
```

Comment:
same nit as above. if you squeeze this onto L99, L99 will still be shorter than L102!